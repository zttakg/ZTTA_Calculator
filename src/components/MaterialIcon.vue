<template>
  <span
    class="material-icon"
    :style="{ fontSize: `${size}px` }"
    v-if="name"
  >
    {{ name }}
  </span>
</template>

<script>
export default {
  name: "MaterialIcon",

  props: {
    name: String,
    size: {
      type: Number,
      default: 24
    }
  }
};
</script>

<style scoped>
@import "https://fonts.googleapis.com/icon?family=Material+Icons";

/**
* Based on Google guideline:
* https://google.github.io/material-design-icons/#icon-font-for-the-web
*/
.material-icon {
  font-family: "Material Icons";
  font-weight: normal;
  font-style: normal;
  font-size: 24px; /* Preferred icon size */
  display: inline-block;
  line-height: 1;
  text-transform: none;
  letter-spacing: normal;
  word-wrap: normal;
  white-space: nowrap;
  direction: ltr;

  /* Support for all WebKit browsers. */
  -webkit-font-smoothing: antialiased;
  /* Support for Safari and Chrome. */
  text-rendering: optimizeLegibility;

  /* Support for Firefox. */
  -moz-osx-font-smoothing: grayscale;

  /* Support for IE. */
  font-feature-settings: "liga";
}
</style>
