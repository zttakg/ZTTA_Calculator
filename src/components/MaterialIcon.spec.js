import { shallowMount } from "@vue/test-utils";
import MaterialIcon from "./MaterialIcon";

describe("<MaterialIcon /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(MaterialIcon);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(MaterialIcon, {
      propsData: {
        name: "test",
        size: 48
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
