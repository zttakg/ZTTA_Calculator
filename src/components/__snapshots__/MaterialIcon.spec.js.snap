// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialIcon /> component should render all props 1`] = `
<span class="material-icon" style="font-size: 48px;">
  test
</span>
`;

exports[`<MaterialIcon /> component should render correct without props 1`] = `undefined`;
